File: mobile/flutter-pos/lib/data/models/role.dart
class Role {
  final int idRole;
  final String namaRole;
  final String? deskripsi;

  const Role({required this.idRole, required this.namaRole, this.deskripsi});

  factory Role.fromJson(Map<String, dynamic> json) => Role(
        idRole: json['id_role'] as int,
        namaRole: json['nama_role'] as String,
        deskripsi: json['deskripsi'] as String?,
      );
}

File: .codex_review_extract/laravel-api/app/Models/Role.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\HasMany;

class Role extends Model
{
    protected $table = 'roles';
    protected $primaryKey = 'id_role';
    public $timestamps = true;

    protected $fillable = [
        'nama_role',
        'deskripsi',
    ];

    public function users(): HasMany
    {
        return $this->hasMany(User::class, 'id_role', 'id_role');
    }
}
